homeassistant/components/mqtt/client.py

```diff
@@ -550,7 +633,7 @@ def no_more_acks() -> bool:
 
         # stop the MQTT loop
         async with self._paho_lock:
-            await self.hass.async_add_executor_job(stop)
+            self._mqttc.disconnect()
```

Comment:
Not sure this is right?

Reply:
I think this makes sense now that the auto reconnect logic is handled in async space